src/main/java/Command.java

```diff
@@ -0,0 +1,210 @@
+            return exitCommand;
+        } else if (command.equals("list")) {
+            return listCommand;
+        } else if (command.contains("done ")) {
+            int task = Integer.parseInt(command.replace("done ", "")) - 1;
+            return new DoneCommand(task);
+        } else if (command.contains("delete ")) {
+            int task = Integer.parseInt(command.replace("delete ", "")) - 1;
+            return new DeleteCommand(task);
+        } else if (command.contains("find ")) {
+            String key = command.replace("find ", "");
+            return new FindCommand(key);
+        } else {
+            //actual entry
+            String postFix = command.split(" ", 2)[1];
+            String preFix = command.split(" ", 2)[0];
+            if (preFix.equals("todo")) {
+                return new TodoCommand(postFix);
+            } else if (preFix.equals("deadline")) {
+                return new DeadlineCommand(postFix);
+            } else if (preFix.equals("event")) {
+                return new EventCommand(postFix);
+            } else {
+                throw new InvalidInput();
+            }
+        }
+    }
+
+    /**
+     * Executes the current command
+     * @param tasks Task list
+     * @param ui User interface
+     * @param storage Storage handling
+     * @throws IOException may be thrown while saving to storage
+     */
+
+    public abstract String execute(TaskList tasks, Ui ui, Storage storage) throws IOException;
+
+    /**
+     * Returns if the command is supposed to cause an exit from UI
+     * @return True/False to exit
+     */
+    public boolean isExit() {
+        return exit;
+    }
+}
+
+/**
+ * An exit command
+ */
+class ExitCommand extends Command {
+    ExitCommand() {
+        super();
+        this.exit = true;
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        System.out.println("EXISTSTAUS");
+        return "Duke is exiting, goodbye!";
+    }
+}
+
+/**
+ * A list command to list all tasks
+ */
+class ListCommand extends Command {
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        System.out.println("tasks:" + tasks.toString());
+        return ui.printf("Here are the tasks in your list:\n" + tasks.toString());
+    }
+}
+
+/**
+ * A done command to mark a given task number as done
+ */
+class DoneCommand extends Command {
+    private int doneTask;
+
+    DoneCommand(int task) {
+        super();
+        this.doneTask = task;
+    }
+
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws IOException {
+        tasks.done(this.doneTask);
+        storage.saveFile(tasks);
+        return ui.printf("Nice! I've marked this task as done:\n" + tasks.get(this.doneTask));
+    }
+}
+
+/**
+ * A delete command to delete a given task number
+ */
+class DeleteCommand extends Command {
+    private int deleteTask;
+
+    DeleteCommand(int task) {
```

Comment:
Can consider renaming the _task_ parameter as _taskIndex_ to make it clearer for users